src/components/ProjectSettingsTabs/GeneralTab/Fields/RepoConfigField.tsx

```diff
@@ -8,65 +8,100 @@ import { SpruceForm } from "components/SpruceForm";
 import { size } from "constants/tokens";
 import { useToastContext } from "context/toast";
 import {
+  AttachProjectToNewRepoMutation,
+  AttachProjectToNewRepoMutationVariables,
   AttachProjectToRepoMutation,
   AttachProjectToRepoMutationVariables,
   DetachProjectFromRepoMutation,
   DetachProjectFromRepoMutationVariables,
 } from "gql/generated/types";
 import {
+  ATTACH_PROJECT_TO_NEW_REPO,
   ATTACH_PROJECT_TO_REPO,
   DETACH_PROJECT_FROM_REPO,
 } from "gql/mutations";
 import { ProjectType } from "../../utils";
 
 interface ModalProps {
-  onCancel: () => void;
-  onConfirm: (formUpdate: any) => void;
+  handleClose: () => void;
   open: boolean;
+  projectId: string;
+  repoName: string;
+  repoOwner: string;
 }
 
 export const MoveRepoModal: React.FC<ModalProps> = ({
-  onCancel,
-  onConfirm,
+  handleClose,
   open,
+  projectId,
+  repoName,
+  repoOwner,
 }) => {
-  const [formState, setFormState] = useState({});
-  const [hasError, setHasError] = useState(false);
+  const dispatchToast = useToastContext();
+
+  const [formState, setFormState] = useState(moveRepoForm.defaultFormData);
+  const [hasError, setHasError] = useState(true);
+
+  const [attachProjectToNewRepo] = useMutation<
+    AttachProjectToNewRepoMutation,
+    AttachProjectToNewRepoMutationVariables
+  >(ATTACH_PROJECT_TO_NEW_REPO, {
+    onCompleted() {
+      dispatchToast.success("Successfully attached to repo");
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error attaching the project: ${err.message}`
+      );
+    },
+    refetchQueries: ["ProjectSettings", "RepoSettings"],
+  });
 
   return (
     <ConfirmationModal
-      buttonText="Move Repo"
+      buttonText="Move Project"
       data-cy="move-repo-modal"
-      onCancel={onCancel}
-      onConfirm={() => onConfirm(formState)}
+      onCancel={handleClose}
+      onConfirm={() => {
+        attachProjectToNewRepo({
+          variables: {
+            project: {
+              projectId,
+              newOwner: formState.owner,
+              newRepo: formState.repo,
+            },
+          },
+        });
+        handleClose();
+      }}
       open={open}
       submitDisabled={hasError}
-      title="Move Repo"
+      title="Move to New Repo"
```

Comment:
I tested this locally and it allowed me to put in a fake owner and repo, and it still gave me a success toast.

Reply:
Right! A feature of of attachProjectTo*New*Repo is that a repo will be created if one does not exist (correct me if I'm wrong @ablack12)

Reply:
ah! should we maybe add that information in a tooltip or help text?

Reply:
It's noted right on the modal where the change is made, so tbh I don't know that it's necessary to call out again for fear of information overload. I'd say instead we should just see if we get any feedback around this modal down the line.

Reply:
my bad, I missed that.